streams/src/test/java/org/apache/kafka/streams/integration/utils/KafkaEmbedded.java

```diff
@@ -63,20 +61,21 @@
 
     /**
      * Creates and starts an embedded Kafka broker.
+     *
      * @param config Broker configuration settings.  Used to modify, for example, on which port the
      *               broker should listen to.  Note that you cannot change the `log.dirs` setting
      *               currently.
      */
-    public KafkaEmbedded(Properties config) throws IOException {
-        tmpFolder = new TemporaryFolder();
-        tmpFolder.create();
-        logDir = tmpFolder.newFolder();
-        effectiveConfig = effectiveConfigFrom(config);
-        boolean loggingEnabled = true;
-        KafkaConfig kafkaConfig = new KafkaConfig(effectiveConfig, loggingEnabled);
+    public KafkaEmbedded(final Properties config) throws IOException {
```

Comment:
For some test cases we may want to use the same `MockTime` object on both server and client, for example in window store changelog truncation (cc @dguy ). So instead of creating the object internally we may want to pass it through parameters.